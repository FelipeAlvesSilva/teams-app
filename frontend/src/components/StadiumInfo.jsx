import React, { Fragment } from 'react'
import { useQuery } from '@apollo/react-hooks'
import { GET_ONE_STADIUM } from '../queries/champioshipsQueries'

const StadiumData = ({ stadium }) => {
    const formattedData = `${stadium.nome} - ${stadium.cidade} ${stadium.estado}`

    return (
        <Fragment>
            <span>{formattedData}</span>
        </Fragment>
    )
}

const StadiumInfo = (props) => {
    const stadium = useQuery(GET_ONE_STADIUM, {
        variables: {
            id: props.idEstadio
        }
    })
    var stadiumInfo = null

    if(stadium.data)
        stadiumInfo = stadium.data.getEstadio[0]

    if (stadium.data)
        return (
            <div>
                <StadiumData stadium={stadiumInfo} />
            </div>
        )
    else
        return null
}

export default StadiumInfo