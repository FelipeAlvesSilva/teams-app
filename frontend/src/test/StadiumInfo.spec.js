import React from 'react'
import renderer from 'react-test-renderer'
import StadiumInfo from '../components/StadiumInfo'
import { MockedProvider } from '@apollo/react-testing'

describe('The display of element', () => {
    it('Should not regress', () => {
        const tree = renderer.create(
            <MockedProvider mocks={[]} addTypename={false}>
                <StadiumInfo idEstadio={'1'} />
            </MockedProvider>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})