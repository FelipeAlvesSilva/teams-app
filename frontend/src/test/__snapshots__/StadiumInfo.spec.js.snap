// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The display of element Should not regress 1`] = `null`;
